EventHandlers/JoinOptinGuildHandler.cs

```diff
@@ -30,12 +30,15 @@ public class JoinOptinGuildHandler : AbstractRegexHandler
 
         /// <summary>
         /// The matcher for detecting the phrases:
-        /// - <@123> join {1}
-        /// - <@123> join #{1}
-        /// And captures the proposed channel name into group 1
+        /// - <@123> join {channel}
+        /// - <@123> join #{channel}
+        /// - <@123> join opt-in {channel}
+        /// - <@123> join opt in {channel}
+        /// - <@123> join optin {channel}
+        /// And captures the proposed channel name into the named capture group "channel"
         /// </summary>
         private static Regex matcher = new Regex(
-            @"^\s*\<\@\!?\d+\>\s*join\s+#?(\S+)$",
+            @"^\s*\<\@\!?\d+\>\s*join(\s+opt(?:-|\s)?in)?\s+#?(?<channel>\S+)$",
```

Comment:
When implementing invite only channels this will cause regex collisions. I'll keep for now in case I find a better way to differentiate. I suppose I could disallow invite channels from having the same name as opt in channels. But then this would leak which invite channels even exist. Hmmmm....